middle-pgsql.cpp

```diff
@@ -75,10 +75,11 @@ namespace {
 // Argument should point to beginning of literal, on return points to delimiter */
```

Comment:
The comments need updating as this function no longer takes arrays, but takes json.

You also need to handle arbitrary JSON inputs. It's acceptable to exit on json which is not a bunch of string key:value mappings, but you need to handle any json within that.

I'm not sure *what* differences that entails, but one obvious one is escaping. JSON allows more escape sequences than just `\n` and `\t`